### ISSUE TITLE
ORDER BY statements are not enclosed by backticks

### ISSUE THREAD

**Issue by joao-coli** [glpi-project/glpi#8226]
**Describe the bug**

When trying to do a Search::show('ITEMXYZ'), the "ORDER BY" generated by the Search class is not enclosed by backticks. It means that something like Search::show("\Ticket") or Search::show("\\\\Ticket"), although both valid, result in a SQL Error.
It's no big deal to native GLPI classes as the backslash is not needed, but it breaks my custom plugin items following the PSR-4 standard.

**Page(s) URL**
--

**To reproduce**

##### In GLPI_ROOT/inc/:
1. Create a file (like 'test.class.php') and put the following:
```
<?php

include_once __DIR__ . "/includes.php";

Search::show("\\Ticket");
```
The same applies to files inside plugins directories, given the right path to the 'includes.php' file.

**Expected behavior**

Search::show('Item') should list items of the item type 'Item', allowing different filters to be made on that list.
In my case, I'd run somehting as follows: Search::show("Foo\\Bar\\PluginXYZItem"). Also, the autoloader for the plugin is enabled and registered, and that command works it the backticks are applied to the ORDER BY statement's column.

**Logs**

```
[2020-11-03 19:22:48] glpisqllog.ERROR: DBmysql::query() in /var/www/html/inc/dbmysql.class.php line 306
  *** MySQL query error:
  SQL: SELECT DISTINCT `glpi_plugin_ramo_containers`.`id` AS id, 'glpi' AS currentuser,
                         `glpi_plugin_ramo_containers`.`nome` AS `ITEM_Inc\Items\PluginRamoContainer_1`,
                        `glpi_plugin_ramo_containers`.`id` AS `ITEM_Inc\Items\PluginRamoContainer_1_id`,
                        `glpi_plugin_ramo_containers`.`email` AS `ITEM_Inc\Items\PluginRamoContainer_2`,  `glpi_plugin_ramo_containers`.`is_active` AS `ITEM_Inc\Items\PluginRamoContainer_3` FROM `glpi_plugin_ramo_containers` ORDER BY ITEM_Inc\Items\PluginRamoContainer_1 ASC  LIMIT 0, 20
  Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '\Items\PluginRamoContainer_1 ASC  LIMIT 0, 20' at line 4
  Backtrace :
  inc/search.class.php:1208                          
  inc/search.class.php:320                           Search::constructData()
  inc/search.class.php:95                            Search::getDatas()
  inc/search.class.php:80                            Search::showList()
  plugins/ramo/front/container.php:65                Search::show()
```

**Screenshots**

Screenshot of the error occurrence:
![image](https://user-images.githubusercontent.com/28827757/98033763-ba509100-1df4-11eb-85f8-2eddf0df653c.png)

**Your GLPI setup (you can find it in Setup > General menu, System tab)**
(Running inside a Docker container)

```
[code]
 
GLPI 9.5.2 ( => /var/www/html)
Installation mode: TARBALL

Server
 
Operating system: Linux 9417fa4e78c3 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64
PHP 7.3.20 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bz2, ctype, curl,
	date, dom, fileinfo, filter, ftp, gd, hash, iconv, intl, json, libxml, mbstring, mysqli, mysqlnd, openssl, pcre, pdo_sqlite,
	posix, session, sodium, sqlite3, standard, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, zip, zlib)
Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="2M" 
Software: Apache/2.4.38 (Debian) (Apache/2.4.38 (Debian) Server at localhost Port 8001)
	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36
Server Software: mariadb.org binary distribution
	Server Version: 10.5.4-MariaDB-1:10.5.4+maria~focal
	Server SQL Mode: 
	Parameters: jcoli@db/glpi
	Host info: db via TCP/IP
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is not presentldap extension is not present
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
CAS extension is not presentCAS extension is not present
exif extension is not presentexif extension is not present
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.5.4) - Perfect!Database version seems correct (10.5.4) - Perfect!
Access to timezone database (mysql) is not allowed.Access to timezone database (mysql) is not allowed.
The log file has been created successfully.The log file has been created successfully.
Write access to /var/www/html/config has been validated.Write access to /var/www/html/config has been validated.
Write access to /var/www/html/files has been validated.Write access to /var/www/html/files has been validated.
Write access to /var/www/html/files/_dumps has been validated.Write access to /var/www/html/files/_dumps has been validated.
Write access to /sessions has been validated.Write access to /sessions has been validated.
Write access to /var/www/html/files/_cron has been validated.Write access to /var/www/html/files/_cron has been validated.
Write access to /var/www/html/files/_graphs has been validated.Write access to /var/www/html/files/_graphs has been validated.
Write access to /var/www/html/files/_lock has been validated.Write access to /var/www/html/files/_lock has been validated.
Write access to /var/www/html/files/_plugins has been validated.Write access to /var/www/html/files/_plugins has been validated.
Write access to /var/www/html/files/_tmp has been validated.Write access to /var/www/html/files/_tmp has been validated.
Write access to /var/www/html/files/_cache has been validated.Write access to /var/www/html/files/_cache has been validated.
Write access to /var/www/html/files/_rss has been validated.Write access to /var/www/html/files/_rss has been validated.
Write access to /var/www/html/files/_uploads has been validated.Write access to /var/www/html/files/_uploads has been validated.
Write access to /var/www/html/files/_pictures has been validated.Write access to /var/www/html/files/_pictures has been validated.
Write access to /var/www/html/marketplace has been validated.Write access to /var/www/html/marketplace has been validated.
Web access to files directory is protectedWeb access to files directory is protected

GLPI constants
 
GLPI_ROOT: /var/www/html
GLPI_CONFIG_DIR: /var/www/html/config
GLPI_VAR_DIR: /var/www/html/files
GLPI_MARKETPLACE_DIR: /var/www/html/marketplace
GLPI_SESSION_DIR: /sessions
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/www/html/files
GLPI_CACHE_DIR: /var/www/html/files/_cache
GLPI_CRON_DIR: /var/www/html/files/_cron
GLPI_DUMP_DIR: /var/www/html/files/_dumps
GLPI_GRAPH_DIR: /var/www/html/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/html/files/_locales
GLPI_LOCK_DIR: /var/www/html/files/_lock
GLPI_LOG_DIR: /var/www/html/files/_log
GLPI_PICTURE_DIR: /var/www/html/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/html/files/_plugins
GLPI_RSS_DIR: /var/www/html/files/_rss
GLPI_TMP_DIR: /var/www/html/files/_tmp
GLPI_UPLOAD_DIR: /var/www/html/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/html/locales
GLPI_VERSION: 9.5.2
GLPI_SCHEMA_VERSION: 9.5.2
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2020

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/var/www/html/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/var/www/html/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.5 in (/var/www/html/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/var/www/html/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/html/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/var/www/html/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/var/www/html/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/var/www/html/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/var/www/html/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/vendor/sebastian/diff/src)
elvanto/litemoji in (/var/www/html/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/html/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/html/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/var/www/html/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/html/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/var/www/html/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/vendor/paragonie/sodium_compat/src)

SQL replicas
 
Not active

Notifications
 
Way of sending emails: PHP

Mails receivers
 

Plugins list
 
	fields               Name: Campos adicionais              Version: 1.12.0     State: Enabled
	gestaohoras          Name: Gestão de Horas                Version: 1.0.0      State: Not installed
	ramo                 Name: Ramo                           Version: 0.9.4.0    State: Enabled

[/code]
```

**Additional context**

I found where to fix this specific issue. In `search.class.php`, line [3197](https://github.com/glpi-project/glpi/blob/9.5/bugfixes/inc/search.class.php#L3197), function addOrderBy, the backticks are missing in several occasions (like in lines 3228 and 3311).

**Comment by trasher** [comment]
Please open a pull request with proposed fix; with that specific case unit tested

**Comment by cedric-anne** [comment]
Fixed in #8234 .
